Ball of Duty Server/Domain/Entities/CharacterFactory.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Ball_of_Duty_Server.Domain.Entities.CharacterSpecializations;
+
+namespace Ball_of_Duty_Server.Domain.Entities
+{
+    public static class CharacterFactory
+    {
+        private static Dictionary<Specializations, Func<Character>> _specializations =
+            new Dictionary<Specializations, Func<Character>>();
+
+
+        static CharacterFactory()
+        {
+            _specializations.Add(Specializations.ROLLER, CharacterFactory.CreateRoller);
+            _specializations.Add(Specializations.BLASTER, CharacterFactory.CreateBlaster);
+            _specializations.Add(Specializations.HEAVY, CharacterFactory.CreateHeavy);
+        }
+
+        public static Character CreateCharacter(Specializations specialization)
+        {
+            Func<Character> f;
+
+            if (!_specializations.TryGetValue(specialization, out f))
```

Comment:
we can eventually do it like this as an exception if the specialization did not exist would already have occured earlier
``return _specializations[specialization]();